storage/files.rb

```diff
@@ -65,6 +77,29 @@ def upload_file project_id:, bucket_name:, local_file_path:,
   # [END upload_file]
 end
 
+def upload_encrypted_file project_id:, bucket_name:, local_file_path:,
+                          storage_file_path: nil, base64_encryption_key:
+  # [START upload_encryption_file]
+  # project_id            = "Your Google Cloud project ID"
+  # bucket_name           = "Your Google Cloud Storage bucket name"
+  # local_file_path       = "Path to local file to upload"
+  # storage_file_path     = "Path to store the file in Google Cloud Storage"
+  # base64_encryption_key = "Base64 encoded AES-256 encryption key"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+
+  bucket = storage.bucket bucket_name
+
+  encryption_key = Base64.decode64 base64_encryption_key
```

Comment:
I'm taking input from the CLI which wouldn't be kind to non-ASCII characters. So I ask for Base64 encoded representation. Is that too much?